### PR TITLE
Memory leak fixes

### PR DESCRIPTION
**Pull request by JLockerman**
This series of commits fix memory leaks across the database.
Note that the PR is still WIP, more commits may be added if more leaks are found.

### CODE REVIEW

File: test/sql/.gitignore [hunk 1]
```diff
@@ -1,10 +1,12 @@
 /agg_bookends-*.sql
+/agg_bookends_optimized-*.sql
```
Comment (svenklemm) [inline test/sql/.gitignore]: this should be removed and the other gitignore change too

File: src/bgw/job.c [hunk 2]
```diff
@@ -613,7 +628,10 @@ ts_bgw_job_entrypoint(PG_FUNCTION_ARGS)
 										TXN_LOCK,
 										/* block */ false);
 		if (job != NULL)
+		{
 			ts_bgw_job_stat_mark_end(job, JOB_FAILURE);
+			pfree(job);
```
Comment (cevian) [inline src/bgw/job.c]: let's set job = null after the pfree here

File: src/bgw/job.c [hunk 3]
```diff
@@ -636,6 +654,9 @@ ts_bgw_job_entrypoint(PG_FUNCTION_ARGS)
 	ts_bgw_job_stat_mark_end(job, res);
 	CommitTransactionCommand();
 
+	if (job != NULL)
+		pfree(job);
```
Comment (cevian) [inline src/bgw/job.c]: also maybe job = NULL here as well?

File: src/bgw/job.c [hunk 4]
```diff
@@ -601,6 +609,13 @@ ts_bgw_job_entrypoint(PG_FUNCTION_ARGS)
 			AbortCurrentTransaction();
 		StartTransactionCommand();
 
+		/* Free the old job if it exists, it's no longer needed, and since it's
+		 * in the TopMemoryContext it won't be freed otherwise.
+		 */
+
+		if (job != NULL)
+			pfree(job);
```
Comment (cevian) [inline src/bgw/job.c]: should we set job = null after the pfree?

**Pull request merged**